runtime/src/main/resources/reference.conf

```diff
@@ -1,13 +1,29 @@
+//#defaults
 akka.grpc.client."*" {
-  host = ""
+  # host to use if service-discovery-mechaism is hardcoded
+  # otherwise the name to look up in service discovery
+  service-name = ""
+
+  # port to use if service-discovery-mechism is hardcoded or service discovery does not return a port
   port = 0
+
+  # timeout for service discovery resolving
   resolve-timeout = 1s
+
   deadline = infinite
   override-authority = ""
   trusted-ca-certificate = ""
   user-agent = ""
   # Pulls default configuration from ssl-config-core's reference.conf
   ssl-config = ${ssl-config}
+
   # TODO: Enforce HTTP/2 TLS restrictions: https://tools.ietf.org/html/draft-ietf-httpbis-http2-17#section-9.2
   connection-attempts = -1
+
+  # Service discovery mechamism to use. The default is to use a hard coded host
+  # and port that will be resolved via DNS.
+  # Any of the mechanisms described [here](https://developer.lightbend.com/docs/akka-management/current/discovery.html) can be used
+  # including Kubernetes, Consul, AWS API
+  service-discovery-mechanism = "hardcoded"
```

Comment:
Could we make it work like this:

```
service-discovery = {
  method = "static"
  host = "xyz"
  port = 0 # or 8080
}
```

or 

```
service-discovery = {
  method = "akka-dns" # or even class name of service discovery
  service-name = "xyz"
}
```

Reply:
static is a much nicer name :+1:

Reply:
nice one, agreed!

Reply:
Also updated to look for host if it is static otherwise service name :+1: